src/types.rs

```diff
@@ -283,6 +284,28 @@ impl Data {
             },
         }
     }
+
+    /// Converts a bytestring to a canonical scalar
+    pub fn to_scalar(self) -> Result<Scalar, VMError> {
+        match self {
+            Data::Opaque(data) => {
+                let scalar = Subslice::new(&data).read_scalar()?;
+                Ok(scalar)
+            }
+            Data::Witness(_) => unimplemented!(),
+        }
+    }
+
+    /// Converts a bytestring to a compressed point
+    pub fn to_point(self) -> Result<CompressedRistretto, VMError> {
```

Comment:
I don't think we need this method. The codepath should be common to the prover and verifier. What we need for `var` instruction is `to_commitment()` which gives a `Commitment` enum.